Investigate flaky Flyway test

There's something fishy with Flyway lately:
```
2022-10-14T23:15:00.4215101Z Caused by: java.lang.RuntimeException: Failed to start quarkus
2022-10-14T23:15:00.4215480Z 	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
2022-10-14T23:15:00.4215875Z 	at io.quarkus.runtime.Application.start(Application.java:101)
2022-10-14T23:15:00.4216406Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2022-10-14T23:15:00.4216956Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
2022-10-14T23:15:00.4217570Z 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
2022-10-14T23:15:00.4218056Z 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
2022-10-14T23:15:00.4218493Z 	at io.quarkus.runner.bootstrap.StartupActionImpl.run(StartupActionImpl.java:237)
2022-10-14T23:15:00.4218960Z 	at io.quarkus.test.QuarkusUnitTest.beforeAll(QuarkusUnitTest.java:642)
2022-10-14T23:15:00.4219274Z 	... 43 more
2022-10-14T23:15:00.4219665Z Caused by: org.flywaydb.core.api.exception.FlywayValidateException: Validate failed: Migrations have failed validation
2022-10-14T23:15:00.4220118Z Migration checksum mismatch for migration version 1.0.0
2022-10-14T23:15:00.4220667Z -> Applied to database : 156834024
2022-10-14T23:15:00.4220964Z -> Resolved locally    : -986920762
2022-10-14T23:15:00.4221294Z Either revert the changes to the migration, or run repair to update the schema history.
2022-10-14T23:15:00.4221716Z Need more flexibility with validation rules? Learn more: https://rd.gt/3AbJUZE
2022-10-14T23:15:00.4222091Z 	at org.flywaydb.core.Flyway.lambda$migrate$0(Flyway.java:134)
2022-10-14T23:15:00.4222714Z 	at org.flywaydb.core.FlywayExecutor.execute(FlywayExecutor.java:204)
2022-10-14T23:15:00.4223102Z 	at org.flywaydb.core.Flyway.migrate(Flyway.java:128)
2022-10-14T23:15:00.4223538Z 	at io.quarkus.flyway.runtime.FlywayRecorder.doStartActions(FlywayRecorder.java:79)
2022-10-14T23:15:00.4224040Z 	at io.quarkus.deployment.steps.FlywayProcessor$startActions1770701860.deploy_0(Unknown Source)
2022-10-14T23:15:00.4224543Z 	at io.quarkus.deployment.steps.FlywayProcessor$startActions1770701860.deploy(Unknown Source)
2022-10-14T23:15:00.4224880Z 	... 51 more
```

/cc @gastaldi if you have an idea.

_Originally posted by @gsmet in https://github.com/quarkusio/quarkus/issues/28613#issuecomment-1279727919_

--- a/extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionRepairAtStartTest.java
+++ b/extensions/flyway/deployment/src/test/java/io/quarkus/flyway/test/FlywayExtensionRepairAtStartTest.java
@@ -26,7 +26,7 @@ public class FlywayExtensionRepairAtStartTest {
             .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
                     .addClass(FlywayResource.class)
                     .addAsResource("db/migration/V1.0.0__Quarkus.sql")
-                    .addAsResource("migrate-at-start-config.properties", "application.properties"))
+                    .addAsResource("repair-at-start-config.properties", "application.properties"))
             .setLogRecordPredicate(r -> true)
             .setAllowFailedStart(true);
 

--- a/extensions/flyway/deployment/src/test/resources/init-sql-config.properties
+++ b/extensions/flyway/deployment/src/test/resources/init-sql-config.properties
@@ -1,7 +1,7 @@
 quarkus.datasource.db-kind=h2
 quarkus.datasource.username=sa
 quarkus.datasource.password=sa
-quarkus.datasource.jdbc.url=jdbc:h2:tcp://localhost/mem:test-quarkus-migrate-at-start;DB_CLOSE_DELAY=-1
+quarkus.datasource.jdbc.url=jdbc:h2:tcp://localhost/mem:init-sql-config;DB_CLOSE_DELAY=-1
 
 # Flyway config properties
 quarkus.flyway.migrate-at-start=true

--- a/extensions/flyway/deployment/src/test/resources/repair-at-start-config.properties
+++ b/extensions/flyway/deployment/src/test/resources/repair-at-start-config.properties
@@ -1,0 +1,7 @@
+quarkus.datasource.db-kind=h2
+quarkus.datasource.username=sa
+quarkus.datasource.password=sa
+quarkus.datasource.jdbc.url=jdbc:h2:tcp://localhost/mem:test-quarkus-repair-at-start;DB_CLOSE_DELAY=-1
+
+# Flyway config properties
+quarkus.flyway.migrate-at-start=true

